### DISCUSSION TITLE
Redirect based on response code

### DISCUSSION THREAD

**Discussion by eadwu** [envoyproxy/gateway#4254] [Q&A] [answered]
Hey, I set up a SecurityPolicy that correctly returns/allows urls. I want to redirect 401 to a login page, from what I could find for routes, I could find a way to match the response. (I see redirect filter and response header)

**Comment by arkodg** [comment, marked as answer]
there's no first class API for this yet in Envoy Gateway

you can achieve this by configuring this filter https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/custom_response_filter#redirect-policy in Envoy directly using https://gateway.envoyproxy.io/docs/tasks/extensibility/envoy-patch-policy/

In the future it could be added into https://github.com/envoyproxy/gateway/issues/1400
